src/mesh/voxel_cloud.rs

```diff
@@ -448,6 +448,165 @@ impl ScalarField {
         tools::weld(&joined_voxel_mesh, (min_voxel_dimension as f32) / 4.0)
     }
 
+    /// Computes a relatively smooth triangulated welded mesh from the current
+    /// state of the scalar field.
+    ///
+    /// For watertight volumetric geometry (i.e. from a watertight source mesh)
+    /// this creates both, outer and inner boundary mesh. There is also a high
+    /// risk of generating a non-manifold mesh.
+    pub fn to_marching_cubes<U>(&self, volume_value_range: &U) -> Option<Mesh>
+    where
+        U: RangeBounds<f32>,
+    {
+        if self.block_dimensions.x == 0
+            || self.block_dimensions.y == 0
+            || self.block_dimensions.z == 0
+            || !self.contains_voxels_within_range(volume_value_range)
+        {
+            return None;
+        }
+
+        let marching_cubes_lookup_table = marching_cubes_lookup_table();
+
+        // A collection of mesh patches defining an outer envelope of volumes
+        // stored in the scalar field
+        let mut marching_cubes: Vec<Mesh> = Vec::new();
```

Comment:
How likely is it that we get an empty result from the marching cubes table? Don't we want to pre-allocate with capacity here?

Reply:
I should never be empty, but we also have no clue about the size.

Reply:
By empty I meant finding an empty row in the lookup table (there are 2), which means we would skip adding it to the mesh patch to this vec.

Reply:
it's very probable - actually it's way more probable than not. only few voxels actually materialize into a mesh, most of them are outside or inside the volumetric shapes.

Reply:
Okay - in this case let's leave it `Vec::new`